### PR TITLE
[MOB-2893] Analytics clean up

### PR DESCRIPTION
**Pull request by lucaschifino**
[MOB-2893]

## Context

We want to delete unused events.

## Approach

When assessing unused events, some unexpected were found. [This spreadsheet](https://docs.google.com/spreadsheets/d/1Xo-zbmp-6pw0mhERucjQ-oguWFC4QMW_me083BjrAqE/edit?pli=1&gid=1504908666#gid=1504908666) was used as an intermediate step to discuss if any of those are a regression bug.

After that, in code, we:
1. Remove the ones that are not relevant
2. Re-add the actual usage of the ones that are still desired
     * Also made sure we add `// Ecosia:` comments as this were not there before the upgrade and are likely the reason we did not find them to add when upgrading

## Other

Took the opportunity to do some additional refactors:
1. Remove Analytics.Values that were not used
2. Leave no hardcoded values on the Analytics methods
3. Reorder things alphabetically and mark sections for easier maintainability
4. Group referrals tracking methods
5. Rename ntp customisation to be more descriptive
6. Split Onboarding specific labels

### Checklist

- [x] I performed some relevant testing on a real device and/or simulator
- [x] ~~I wrote Unit Tests that confirm the expected behaviour~~ No unit tests atm, will investigate that and add in a follow-up if a good way is found
- [x] I added the `// Ecosia:` helper comments where needed
- [x] I made sure that any change to the Analytics events included in PR won't alter current analytics (e.g. new users, upgrading users)

[MOB-2893]: https://ecosia.atlassian.net/browse/MOB-2893?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Client/Ecosia/Analytics/Analytics.Values.swift [hunk 1]
```diff
@@ -4,125 +4,173 @@ extension Analytics {
     enum Category: String {
         case
         activity,
-        abTest = "ab_Test",
```
Comment (d4r1091) [inline Client/Ecosia/Analytics/Analytics.Values.swift:7]: Isn't this needed anymore?

File: Client/Ecosia/Extensions/BrowserViewController+Ecosia.swift [hunk 2]
```diff
@@ -31,19 +31,19 @@ extension BrowserViewController: PageActionsShortcutsDelegate {
     func pageOptionsOpenHome() {
         tabToolbarDidPressHome(toolbar, button: .init())
         dismiss(animated: true)
-        Analytics.shared.menuClick("home")
+        Analytics.shared.menuClick(.home)
```
Comment (d4r1091) [inline Client/Ecosia/Extensions/BrowserViewController+Ecosia.swift:34]: 👏

**Review by d4r1091 — COMMENTED** [review]
Really good refactor. 👏 
Well done!
I've got only 1 clarification regarding `ab_test`.
Furthermore, I saw a few brand-new additions. Does that mean that weren't  tracked at all?
Thanks!

File: Client/Ecosia/Analytics/Analytics.Values.swift (hunk 1 above)
Comment (lucaschifino) [reply Client/Ecosia/Analytics/Analytics.Values.swift:7]: It's not being used anywhere in code, so no reason to keep it imo. If we ever need it again we can re-add.

**Comment by lucaschifino** [comment]
> Furthermore, I saw a few brand-new additions. Does that mean that weren't tracked at all?

There were a number of events that were missing after the last upgrade, they were not properly commented with Ecosia and therefore were very easily missed. You can see the discussion with regards to those on the ticket comments. I also used the [Mobile Events spreadsheet](https://docs.google.com/spreadsheets/d/1Xo-zbmp-6pw0mhERucjQ-oguWFC4QMW_me083BjrAqE/edit?pli=1&gid=1504908666#gid=1504908666) to keep track of the changes.

This is also the main reason there's the [testing follow-up ticket](https://ecosia.atlassian.net/browse/MOB-2979).

**Review by d4r1091 — APPROVED** [review]
All good my side! Thanks for the clarifications!

**Pull request merged**